The `pants` script no longer works with 1.30.x

Failure looks like so (here trying to run Pants 1.30.1)
```
AssertionError: Bootstrapping Pants using /home/jsirois/.pyenv/shims/python3.6
  Creating the virtualenv PEX.
  Downloading the Pex PEX.
    % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                   Dload  Upload   Total   Spent    Left  Speed
  
    0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  100   640  100   640    0     0   1774      0 --:--:-- --:--:-- --:--:--  1777
  
    0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  100 3609k  100 3609k    0     0  2810k      0  0:00:01  0:00:01 --:--:-- 3953k
  SHA256 fingerprint of https://github.com/pantsbuild/pex/releases/download/v2.1.62/pex verified.
  WARNING: Retrying (Retry(total=4, connect=None, read=None, redirect=None, status=None)) after connection broken by 'ReadTimeoutError("HTTPSConnectionPool(host='pypi.org', port=443): Read timed out. (read timeout=15.0)",)': /simple/filelock/
  Installing pantsbuild.pants==1.30.5rc1 into a virtual environment at /tmp/pytest-of-jsirois/pytest-7/test_only_bootstraps_the_first0/PANTS_SETUP_CACHE/bootstrap-Linux-x86_64/1.30.5rc1_py36
  created virtual environment CPython3.6.15.final.0-64 in 90ms
    creator CPython3Posix(dest=/tmp/pytest-of-jsirois/pytest-7/test_only_bootstraps_the_first0/PANTS_SETUP_CACHE/bootstrap-Linux-x86_64/pants.dEsSDq/install, clear=False, no_vcs_ignore=False, global=False)
    seeder FromAppData(download=False, pip=bundle, setuptools=bundle, wheel=bundle, via=copy, app_data_dir=/home/jsirois/.local/share/virtualenv)
      added seed packages: pip==21.3.1, setuptools==59.6.0, wheel==0.36.2
    activators BashActivator,CShellActivator,FishActivator,PowerShellActivator,PythonActivator,XonshActivator
  Collecting pantsbuild.pants==1.30.5rc1
    Using cached pantsbuild.pants-1.30.5rc1-cp36-cp36m-manylinux2014_x86_64.whl (47.8 MB)
  Collecting typing-extensions==3.7.4.2
    Using cached typing_extensions-3.7.4.2-py3-none-any.whl (22 kB)
  Collecting setuptools==44.0.0
    Using cached setuptools-44.0.0-py2.py3-none-any.whl (583 kB)
  Collecting twitter.common.dirutil<0.4,>=0.3.11
    Using cached twitter.common.dirutil-0.3.11.tar.gz (9.5 kB)
    Preparing metadata (setup.py): started
    Preparing metadata (setup.py): finished with status 'done'
  Collecting pex==2.1.24
    Using cached pex-2.1.24-py2.py3-none-any.whl (2.5 MB)
  Collecting dataclasses==0.6
    Using cached dataclasses-0.6-py3-none-any.whl (14 kB)
  Collecting Pygments==2.6.1
    Using cached Pygments-2.6.1-py3-none-any.whl (914 kB)
  Collecting toml==0.10.1
    Using cached toml-0.10.1-py2.py3-none-any.whl (19 kB)
  Collecting requests[security]>=2.20.1
    Using cached requests-2.27.1-py2.py3-none-any.whl (63 kB)
  Collecting PyYAML<5.4,>=5.3.1
    Using cached PyYAML-5.3.1.tar.gz (269 kB)
    Preparing metadata (setup.py): started
    Preparing metadata (setup.py): finished with status 'done'
  Collecting pathspec==0.8.0
    Using cached pathspec-0.8.0-py2.py3-none-any.whl (28 kB)
  Collecting packaging==20.3
    Using cached packaging-20.3-py2.py3-none-any.whl (37 kB)
  Collecting Markdown==2.1.1
    Using cached Markdown-2.1.1.zip (341 kB)
    Preparing metadata (setup.py): started
    Preparing metadata (setup.py): finished with status 'done'
  Collecting ansicolors==1.1.8
    Using cached ansicolors-1.1.8-py2.py3-none-any.whl (13 kB)
  Collecting www-authenticate==0.9.2
    Using cached www-authenticate-0.9.2.tar.gz (2.4 kB)
    Preparing metadata (setup.py): started
    Preparing metadata (setup.py): finished with status 'done'
  Collecting python-Levenshtein==0.12.1
    Using cached python-Levenshtein-0.12.1.tar.gz (50 kB)
    Preparing metadata (setup.py): started
    Preparing metadata (setup.py): finished with status 'done'
  Collecting pystache==0.5.4
    Using cached pystache-0.5.4.tar.gz (75 kB)
    Preparing metadata (setup.py): started
    Preparing metadata (setup.py): finished with status 'error'
    ERROR: Command errored out with exit status 1:
     command: /tmp/pytest-of-jsirois/pytest-7/test_only_bootstraps_the_first0/PANTS_SETUP_CACHE/bootstrap-Linux-x86_64/pants.dEsSDq/install/bin/python -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-pz56_4vk/pystache_fac0619cc279482e866112764b0be8c4/setup.py'"'"'; __file__='"'"'/tmp/pip-install-pz56_4vk/pystache_fac0619cc279482e866112764b0be8c4/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /tmp/pip-pip-egg-info-r4qf9kxc
         cwd: /tmp/pip-install-pz56_4vk/pystache_fac0619cc279482e866112764b0be8c4/
    Complete output (2 lines):
    pystache: using: version '59.6.0' of <module 'setuptools' from '/tmp/pytest-of-jsirois/pytest-7/test_only_bootstraps_the_first0/PANTS_SETUP_CACHE/bootstrap-Linux-x86_64/pants.dEsSDq/install/lib/python3.6/site-packages/setuptools/__init__.py'>
    error in pystache setup command: use_2to3 is invalid.
    ----------------------------------------
  WARNING: Discarding https://files.pythonhosted.org/packages/d6/fd/eb8c212053addd941cc90baac307c00ac246ac3fce7166b86434c6eae963/pystache-0.5.4.tar.gz#sha256=f7bbc265fb957b4d6c7c042b336563179444ab313fb93a719759111eabd3b85a (from https://pypi.org/simple/pystache/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
  ERROR: Could not find a version that satisfies the requirement pystache==0.5.4 (from pantsbuild-pants) (from versions: 0.1.0, 0.2.0, 0.3.0, 0.3.1, 0.4.0, 0.4.1, 0.5.0, 0.5.1, 0.5.2, 0.5.3, 0.5.4, 0.6.0)
  ERROR: No matching distribution found for pystache==0.5.4
```

--- a/pants
+++ b/pants
@@ -343,7 +343,9 @@ function bootstrap_pants {
         scrub_PEX_env_vars
         # shellcheck disable=SC2086
         "${python}" "${virtualenv_path}" --no-download "${staging_dir}/install" && \
-        "${staging_dir}/install/bin/pip" install -U pip && \
+        # Grab the latest pip, but don't advance setuptools past 58 which drops support for the
+        # `setup` kwarg `use_2to3` which Pants 1.x sdist dependencies (pystache) use.
+        "${staging_dir}/install/bin/pip" install -U pip "setuptools<58" && \
         "${staging_dir}/install/bin/pip" install ${maybe_find_links} --progress-bar off "${pants_requirement}"
       ) && \
       ln -s "${staging_dir}/install" "${staging_dir}/${target_folder_name}" && \

